### PR TITLE
Add guards to unguarded traces

### PR DESCRIPTION
**Pull request by MrTrader8**
In the code there are places where unguarded tracing occurs causing
an output into a log file when no trace flags are specified.
This commit adds guards to stop the unguarded tracing.

Signed-off-by: Andrew Gao <andrewgao98@gmail.com>

### CODE REVIEW

File: runtime/compiler/optimizer/DataAccessAccelerator.hpp [hunk 1]
```diff
@@ -243,14 +243,15 @@ class TR_DataAccessAccelerator : public TR::Optimization
 
    bool printInliningStatus(bool status, TR::Node* node, const char* reason = "")
       {
-      if (status)
-         traceMsg(comp(), "DataAccessAccelerator: Intrinsics on node %p : SUCCESS\n", node);
-      else
+      if (trace()){
```
Comment (andrewcraik) [inline runtime/compiler/optimizer/DataAccessAccelerator.hpp]: please match code formatting in surrounding code for the {}s

File: runtime/compiler/optimizer/SPMDParallelizer.cpp [hunk 2]
```diff
@@ -3180,12 +3180,14 @@ bool TR_SPMDKernelParallelizer::processGPULoop(TR_RegionStructure *loop, TR_SPMD
    TR::Node *lastNode = loopInvariantBlock->getLastRealTreeTop()->getNode();
    if (lastNode->getOpCode().isGoto())
       {
+      if (trace())
       traceMsg(comp(), "Changing destination of goto at the end of block_%d\n", loopInvariantBlock->getNumber());
```
Comment (andrewcraik) [inline runtime/compiler/optimizer/SPMDParallelizer.cpp]: please indent traceMsg

**Comment by andrewcraik** [comment]
The chatter from SPMDPreCheck is pretty important because it will prevent a number of loop transformations leading up to SPMDParallization. I am of the opinion these should probably remain chatting - there are not many messages since it is at most one per loop.

I'm not sure I am a fan of passing the whole CISCTransformer around in IdiomRecognition to test if tracing is enabled. Wouldn't it be cleaner to have an argument holding the tracing state to avoid extra dependencies being added to the transformer?

**Comment by MrTrader8** [comment]
@andrewcraik I have made the changes to the formatting of the code, as for SPMDPreCheck, guards were added since they were outputting quite a bit of lines(over 50+), however I can remove the guards if needed. In the case for CISCTransformer being passed around, would passing in disptrace as an additional boolean parameter to keep track of tracing be a better option?

**Comment by andrewcraik** [comment]
I personally would prefer the bool because it would stop additional dependencies onto the transformer being introduced. For SPMDPreCheck I still think the output is useful in a traceFull scenario because it tells you what loops we gave up on and won't look at further...

**Comment by MrTrader8** [comment]
@andrewcraik I have changed the functions to pass in a bool instead of CISCTransformer to keep track of the tracing, additionally for the SPMDPreCheck I have changed the tracing guards to print in a TraceFull scenario.

File: runtime/compiler/optimizer/IdiomTransformations.cpp [hunk 3]
```diff
@@ -1086,8 +1096,9 @@ findIndVarLoads(TR::Node *node, TR::Node *indVarStoreNode, bool &storeFound,
    }
 
 static int32_t
-checkForPostIncrement(TR::Compilation *comp, TR::Block *loopHeader, TR::Node *loopCmpNode, TR::Symbol *ivSym)
+checkForPostIncrement(TR_CISCTransformer *trans, TR::Compilation *comp, TR::Block *loopHeader, TR::Node *loopCmpNode, TR::Symbol *ivSym)
```
Comment (andrewcraik) [inline runtime/compiler/optimizer/IdiomTransformations.cpp]: Was this one going to be updated with a bool as well?

File: runtime/compiler/optimizer/SPMDPreCheck.cpp [hunk 4]
```diff
@@ -29,11 +29,13 @@
 
 bool SPMDPreCheck::isSPMDCandidate(TR::Compilation *comp, TR_RegionStructure *loop)
    {
- 
+   bool trace = comp->getOption(TR_TraceAll);
```
Comment (andrewcraik) [inline runtime/compiler/optimizer/SPMDPreCheck.cpp]: Shoudl this be `comp->getOption(TR_TraceALL) || trace()` to make sure the trace turns on if only SPMDPreCheck is being traced?

**Comment by gita-omr** [comment]
@MrTrader8 do you recall what prevented us from completing this PR?

**Comment by MrTrader8** [comment]
@gita-omr I believe this PR was just not merged, if i recall correctly the omr side of the PR was merged already.

**Comment by gita-omr** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by gita-omr** [comment]
@MrTrader8 most likely you will need to rebase your changes. Let's try to merge it soon.

**Comment by gita-omr** [comment]
This work became a part of #9586, so  closing.

**Pull request closed without merging**